src/plugins/search/component.js

```diff
@@ -174,7 +174,9 @@ function bindEvents() {
   Docsify.dom.on(
     $search,
     'click',
-    e => e.target.tagName !== 'A' && e.stopPropagation()
+    e =>
+      ['A', 'H2', 'P', 'EM'].indexOf(e.target.tagName) === -1 &&
```

Comment:
what about other tags ?

Reply:
Are there any others? All i see is this.
![image](https://user-images.githubusercontent.com/33931153/81785904-7fe6f500-9531-11ea-8f0e-5378587dd7b3.png)

Reply:
ahh, yeah I checked the code, it seems these are enough. we can add them later if we missed something 👍

Reply:
What about using `preventDefault` instead of `stopPropagation`? It would make the event handleable by high up code.

Also, maybe we can make this more generic and not maintain a list of specific DOM nodes, so that code won't break if we update markup structure.

How about something like 

```js
// select any anchor and anything inside the anchors
const linkContent = $search.find('a *')

// ...

    e =>
      !$(e.target).is(linkContent) && e.preventDefault()?
```

?

Reply:
In general I think we should improve output so Docsify content is clear, and we can rely on something like

```js
const linkContent = $search.find('.docsify-link *')
```

I had started doing that in my one messy PR. I will split it up soon. But feel free to pluck changes if I'm too slow!

Reply:
Oh, I think `stopPropagation` is to prevent the menu from closing in another event handler, isn't it? Nevermind preventDefault then (for now).

In that case, it may be cleaner to consolidate that other code with this code, so that the intent is clear, but that's a bigger task.

Reply:
Today I also found this code :) I have question, what if we totally delete this section? 
```

// Prevent to Fold sidebar   -   what does it mean???
    Docsify.dom.on(
      $search,
      'click',
      function (e) { return e.target.tagName !== 'A' && e.stopPropagation(); }
    );
```

I deleted it and nothing happens. What bug this code fix? Do you know? I think better will delete this code instead try determine which tag type user click.

Reply:
> 今天，我也找到了这段代码：)我有一个问题，如果我们完全删除此部分该怎么办？
> 
> ```
> 
> // Prevent to Fold sidebar   -   what does it mean???
>     Docsify.dom.on(
>       $search,
>       'click',
>       function (e) { return e.target.tagName !== 'A' && e.stopPropagation(); }
>     );
> ```
> 
> 我删除了它，没有任何反应。此代码修复了哪些错误？你知道吗？我认为最好删除此代码，而不是尝试确定用户单击哪种标记类型。

When searching on the mobile end, the sidebar is collapsed when you click the INPUT box, making it impossible to search.

Reply:
> Today I also found this code :) I have question, what if we totally delete this section?
> 
> ```
> 
> // Prevent to Fold sidebar   -   what does it mean???
>     Docsify.dom.on(
>       $search,
>       'click',
>       function (e) { return e.target.tagName !== 'A' && e.stopPropagation(); }
>     );
> ```
> 
> I deleted it and nothing happens. What bug this code fix? Do you know? I think better will delete this code instead try determine which tag type user click.

When searching on the mobile end, the sidebar is collapsed when you click the INPUT box, making it impossible to search.

Reply:
@trusktr Hi, I think it's okay to modify it this way for now, and there should be no other DOM.
If it's going to be modified, it should be in v5, and the next release shouldn't be v5.

Reply:
In this case I suggest:

1. Write commend that describe why this code needed. For example:
```
When searching on the mobile end, the sidebar is collapsed when you click the INPUT box, making it impossible to search.
```

2. I suggest try to search closest parent link `A` instead all of this tags. Maybe it will be more correct? 

What do you think about it?

Reply:
@stefek99 As I said in https://github.com/docsifyjs/docsify/pull/1175#discussion_r424240260, there are only so many DOMs... 
if there are other more excellent way you can modify this PR, thanks!

Reply:
@stepozer Thanks to the first point of suggestion, which has been revised. 

On the second point, as I said in https://github.com/docsifyjs/docsify/pull/1175#discussion_r424240260, there are only so many DOMs... 

if there are other more excellent way you can modify this PR, thanks!

Reply:
@sy-records as I suggested you can:

1. stopPropagation if current element == `A` tag
2. stopPropagation if current element has parent == `A` tag

Because how I understand all this tags always wrapped in `A`. What do you think?

Reply:
`stopPropagation if current element == A tag` This doesn't seem right. 🤔

Reply:
For pure JS exists equivalent https://stackoverflow.com/questions/38469830/what-is-the-pure-javascript-equivalent-of-jquerys-parents-method

Reply:
> `stopPropagation if current element == A tag` This doesn't seem right.

Sorry, I mean `!==`. Right now you has logic:

if  e.target.tagName !== 'A' then stopPropagation. But in your tag `A` exists other tags, so you need check parent for this tag:

stopPropagation if current element !== A AND current element has NO parents with A tag.

Reply:
I tried write code, I think it will be something like this:

```
function getParentByTag(e, parentTag) {
    for (var p = e && e.parentElement; p; p = p.parentElement) {
      if (p && p.tagName == parentTag) {
        return p;
      }
    }
    return null;
  }

  function elementIsLinkOrHasLinkOnParents(element) {
    var parenLink = getParentByTag(element, 'A')
    return element.tagName === 'A' || parenLink != null;
  }

// Prevent to Fold sidebar
Docsify.dom.on(
  $search,
  'click',
  function (e) { return !elementIsLinkOrHasLinkOnParents(e.target) && e.stopPropagation(); }
);
```

Sorry, I am back-end developer :)

Reply:
I tested on mobile, seems to be working...

Reply:
It's a little complicated... Let's keep it simple. 😥

Reply:
I will create separated PR for this :) I think it will be more right way

Reply:
Sure, feel free to submit a PR. 👍🏻

Reply:
Will wait until your PR will be merged